src/store.js

```diff
@@ -94,7 +94,11 @@ class Store {
     }
 
     if (_.isArray(data)) {
-      return new Collection(data.map(model => this._pushInternalModel(model)))
+      const result = new Collection(data.map(model => this._pushInternalModel(model)));
+      if (meta) {
+        result.meta = this._parser._parseWithNames(meta);
+      }
+      return result;
```

Comment:
This looks like it works perfectly fine but have you looked into extending the Backbone.Collection? It follows the Backbone convention a little more I think. You could even add your own on-demand method.

```suggestion
      const internalModelCollection = Collection.extend({
        meta: null,
        getMeta: () => {
          if (!this.meta && meta) {
            this.meta = this._parser._parseWithNames(meta)
          }
          return this.meta
        }
      })
      return new internalModelCollection(data.map(model => this._pushInternalModel(model)))
```

Reply:
I didn't think about that, but I'd personally prefer to access "meta"  as a POJO. So, what do you think about the following?:

```suggestion
      if (meta) {
        meta = this._parser._parseWithNames(meta);
      }
      const internalModelCollection = Collection.extend({ meta });
      return new internalModelCollection(data.map(model => this._pushInternalModel(model)))
```
Unless you're keen on delaying the work of the parser until the client uses it. Also for the above code, is `meta` in scope for the collection after it is returned?

Reply:
In that case I would stick with your initial solution. Backbone prefers methods (get, set, has, etc) and I think only extending `meta` will keep it private and inaccessible. You wouldn't be able to access it with `collection.meta` as if you would with your initial solution.